File: internal/pkg/database/models/storage_drives_groups.go
package models

import (
	"time"

	"github.com/gofrs/uuid/v5"
)

type StorageDrivesGroups struct {
	StorageDrivesID   []uuid.UUID `json:"storage_drives_id,omitempty"`
	DirectoryGroupsID []uuid.UUID `json:"directory_groups_id,omitempty"`
	Description       []string    `json:"description,omitempty"`
	CreatedOn         []time.Time `json:"created_on,omitempty"`
	LastUpdatedOn     []time.Time `json:"last_updated_on,omitempty"`
	DeactivatedOn     []time.Time `json:"deactivated_on,omitempty"`
}

type storageDrivesGroupsTable struct {
	TableName string

	StorageDrivesID   string
	DirectoryGroupsID string
	Description       string
	CreatedOn         string
	LastUpdatedOn     string
	DeactivatedOn     string
}

func StorageDrivesGroupsTable() storageDrivesGroupsTable {
	return storageDrivesGroupsTable{
		TableName: "storage_drives_groups",

		StorageDrivesID:   "storage_drives_id",
		DirectoryGroupsID: "directory_groups_id",
		Description:       "description",
		CreatedOn:         "created_on",
		LastUpdatedOn:     "last_updated_on",
		DeactivatedOn:     "deactivated_on",
	}
}

File: web/src/lib/database/storage_drives_groups.ts
namespace _StorageDrivesGroups {
	export enum Column {
		StorageDrivesID = 'storage_drives_id',
		DirectoryGroupsID = 'directory_groups_id',
		Description = 'description',
		CreatedOn = 'created_on',
		LastUpdatedOn = 'last_updated_on',
		DeactivatedOn = 'deactivated_on'
	}

	export const TableName = 'storage_drives_groups'

	export interface Interface {
		storage_drives_id?: string[]
		directory_groups_id?: string[]
		description?: string[]
		created_on?: string[]
		last_updated_on?: string[]
		deactivated_on?: string[]
	}
}

export default _StorageDrivesGroups
